releases_test.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"bufio"
+	"os"
+	"testing"
+
+	"github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/versionbundle"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/apiserver/validation"
+	"sigs.k8s.io/yaml"
+)
+
+// sigs.k8s.io/yaml can't handle multi-part YAML documents (separated by "---")
+// so documents such as aws.yaml must be split on boundaries and parsed
+// individually.
+func scanDocuments(filename string) ([]string, error) {
+	data, err := os.Open(filename)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	scanner := bufio.NewScanner(data)
+	var document string
+	var documents []string
+	for scanner.Scan() {
+		line := scanner.Text()
+		if line != "---" {
+			document += line + "\n"
+			continue
+		}
+		if document == "" {
+			continue
+		}
+		documents = append(documents, document)
+		document = ""
+	}
+	return documents, nil
+}
+
+// Given a slice of strings defining Release CRs as YAML, this function
+// simply parses the YAML and returns a slice of Releases.
+func parseReleases(documents []string) ([]v1alpha1.Release, error) {
+	var releases []v1alpha1.Release
+	for _, document := range documents {
+		var release v1alpha1.Release
+		err := yaml.Unmarshal([]byte(document), &release)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+		releases = append(releases, release)
+	}
+	return releases, nil
+}
+
+// To reuse versionbundle.ValidateIndexReleases, the slice of Releases must first be
+// converted into a slice of versionbundle.IndexRelease.
+func releasesToIndex(releases []v1alpha1.Release) []versionbundle.IndexRelease {
```

Comment:
Do we still need the versionbundle package? Is there more we can cleanup?

Reply:
I feel that `versionbundle` is in the process of being deprecated. In this case, it's providing a valuable feature of ensuring that releases are different in at least one component. I can copy that code here (not that much to it I imagine) and probably take it out of `versionbundle` in a separate PR as long as it's not used elsewhere.